src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2807,68 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
 #endif
 }
 
+TEST_CASE("LedgerKeyMeter tests")
+{
+    LedgerKeyMeter lkMeter{};
+    auto entries = LedgerTestUtils::generateValidLedgerEntriesWithExclusions(
+        {OFFER, DATA, CLAIMABLE_BALANCE, LIQUIDITY_POOL, CONFIG_SETTING, TTL},
+        10);
+    // If the key has not been added to the LedgerKeyMeter, it should have a
+    // maximum read quota.
+    REQUIRE(lkMeter.maxReadQuotaForKey(
+                LedgerEntryKey(const_cast<const LedgerEntry&>(entries[0]))) ==
+            std::numeric_limits<uint32_t>::max());
+
+    for (size_t i = 0; i < entries.size(); ++i)
```

Comment:
I don't think we need to call this in a loop, I think you can just run all these tests once since the inner body is just dealing with a single TX and key at a time.